src/passes/function_passes/CheckPointPass.cpp

```diff
@@ -104,30 +105,68 @@ struct CheckPointPass: public FunctionPass {
             intrinsic = Intrinsic::getDeclaration(
                 mod, Intrinsic::experimental_stackmap);
           }
-          auto liveVariables = getLiveRegisters(*it);
-          // Pass the live locations to the stackmap/patchpoint call. This also
-          // inserts the size of each 'live' location into the stackmap.
-          std::move(liveVariables.begin(), liveVariables.end(),
-                    std::back_inserter(args));
-          auto call_inst = builder.CreateCall(intrinsic, args);
+          auto callInst = builder.CreateCall(intrinsic, args);
         } else if (isa<CallInst>(it)) {
           // Insert a stackmap call after each function call inside which a
           // guard may fail. This enables the runtime to work out the return
           // address of the function.
-          Function *calledFun = cast<CallInst>(*it).getCalledFunction();
-          if (!calledFun->hasAvailableExternallyLinkage() &&
+          CallInst &oldCallInst = cast<CallInst>(*it);
+
+          Function *calledFun = oldCallInst.getCalledFunction();
+          if (!oldCallInst.isInlineAsm() &&
+              !calledFun->hasAvailableExternallyLinkage() &&
               !calledFun->isDeclaration()) {
             uint64_t PPID = getNextPatchpointID(funName);
             IRBuilder<> builder(&bb, ++it);
             auto args = vector<Value*> { builder.getInt64(PPID),
-                                         builder.getInt32(13)  // XXX shadow
+                                         builder.getInt32(13)
                                        };
-            auto liveVariables = getLiveRegisters(*it);
-            std::move(liveVariables.begin(), liveVariables.end(),
-                      std::back_inserter(args));
-            auto intrinsic = Intrinsic::getDeclaration(
-                mod, Intrinsic::experimental_stackmap);
-            auto call_inst = builder.CreateCall(intrinsic, args);
+            if (funName.startswith(UNOPT_PREFIX)) {
```

Comment:
I'm finding this block of code a little opaque. Would it benefit from some comments?